app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -10,130 +10,169 @@ import SupplementaryInformationRoute from './SupplementaryInformationRoute'
 import DownloadsRoute from './DownloadsRoute'
 
 const RoutePropTypes = {
-  match: PropTypes.object.isRequired,
-  location: PropTypes.object.isRequired,
-  history: PropTypes.object.isRequired
+    match: PropTypes.object.isRequired,
+    location: PropTypes.object.isRequired,
+    history: PropTypes.object.isRequired
 }
 
 const TabCommonPropTypes = {
-  atlasUrl: PropTypes.string.isRequired,
-  experimentAccession: PropTypes.string.isRequired,
-  species: PropTypes.string.isRequired,
-  accessKey: PropTypes.string,
-  resourcesUrl: PropTypes.string
+    atlasUrl: PropTypes.string.isRequired,
+    experimentAccession: PropTypes.string.isRequired,
+    species: PropTypes.string.isRequired,
+    accessKey: PropTypes.string,
+    resourcesUrl: PropTypes.string
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
-const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
-}
+let tabTypeComponent = []
 
-const TopRibbon = ({tabs, routeProps}) =>
-  <ul className={`tabs`}>
-    {
-      tabs.map((tab) =>
-        <li title={tab.name} key={tab.type} className={`tabs-title`}>
-          <NavLink to={{pathname:`/${tab.type}`, search: routeProps.location.search, hash: routeProps.location.hash}}
-                   activeClassName={`active`}>
-            {tab.name}
-          </NavLink>
-        </li>
-      )}
-  </ul>
+function isThisTabType(tab, tabType) {
+    return tab.type === tabType;
+}
 
+const isObjectEmpty = (objectName) => {
+    return (
+        objectName &&
+        Object.keys(objectName).length === 0 &&
+        objectName.constructor === Object
+    );
+};
+
+const tabTypes = [
+    { type: 'results', key: 'ks', component: TSnePlotViewRoute, optionsKey: 'plotTypesAndOptions' },
+    { type: 'experiment-design', key: 'table.data', component: ExperimentDesignRoute },
+    { type: 'supplementary-information', key: 'sections', component: SupplementaryInformationRoute },
+    { type: 'resources', key: 'data', component: DownloadsRoute }
+];
+
+const getNestedProperty = (obj, path) => path.split('.').reduce((acc, key) => acc?.[key], obj);
+
+const isNonEmptyArray = (value) => !isObjectEmpty(value) && Array.isArray(value);
```

Comment:
Should we move this code: `Array.isArray(value)`  into its relevant `proptype` definition?

Reply:
This is one of the validation checks before displaying a relevant tab. We are getting empty for some experiments.